devtools/server/actors/replay/auth.js

```diff
@@ -117,6 +107,29 @@ function tokenExpiration(token) {
 }
 
 let authChannel;
+// Tracks the open replay.io tabs. If one is closed, its currentWindowGlobal
+// will be set to null and will be removed from the map on the next pass
+const webChannelTargets = new Map();
+
+// Notifies all replay.io tabs of the current auth state
+function notifyWebChannelTargets() {
+  if (authChannel) {
+    for (let [key, target] of webChannelTargets.entries()) {
+      if (target.browsingContext.currentWindowGlobal) {
+        notifyWebChannelTarget(target);
+      } else {
+        webChannelTargets.delete(key)
```

Comment:
This will delete the Map entry during iteration meaning I have no idea what this `for` loop will do.
```diff
-for (let [key, target] of webChannelTargets.entries()) {
+for (let [key, target] of Array.from(webChannelTargets.entries())) {
```
please!

Reply:
I did verify that the iterator is okay with deleting entries while iterating so I don't think this is necessary.

Reply:
Hmm, I guess you're technically right, I just find this terrifying. Any mutation of something during iteration should be avoided in my opinion.

Reply:
This is a side note but I feel like an under-rated solution to these kinds of problems is iterating backwards 😛